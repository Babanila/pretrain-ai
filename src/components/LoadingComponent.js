import React from "react";
import { cx, css } from "emotion";

function LoadingComponent() {
  const loadingStyle = css`
    width: 100%;
    height: 10em;
    display: flex;
    justify-content: center;
    align-items: center;
    color: #f54b3a;
    margin-top: 10em;
  `;
  return <h3 className={cx(loadingStyle)}>Loading ...</h3>;
}

export default LoadingComponent;
