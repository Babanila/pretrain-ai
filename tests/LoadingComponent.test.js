import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import LoadingComponent from '../src/components/LoadingComponent'

describe('Check LoadingComponent Component', () => {
  it('should renders without crashing', () => {
    const tree = renderer.create(<LoadingComponent />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render correct number of element in LoadingComponent component', () => {
    const LoadingComponentWrapper = shallow(<LoadingComponent />)
    const actual = LoadingComponentWrapper.find('h3')

    expect(LoadingComponent).toBeDefined()
    expect(actual.exists).toBeTruthy()
    expect(actual.length).toEqual(1)
    expect(actual.text()).toEqual('Loading ...')
  })
})
