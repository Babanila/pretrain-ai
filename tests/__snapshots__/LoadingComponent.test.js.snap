// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check LoadingComponent Component should renders without crashing 1`] = `
<h3
  className="css-d8gwku"
>
  Loading ...
</h3>
`;
